test/test_sparse.py

```diff
@@ -1611,6 +1611,74 @@ def test_shape(i_shapes, v_shapes, nnzs):
         test_shape([0, 3, 4], [3, 4, 5, 6], [0])
         test_shape([2, 3, 4], [0, 4, 5, 6], [9, 12])
 
+        sparse_tensor, _, _ = self._gen_sparse(len([2, 3, 4]), 9, [2, 3, 4] + [3, 4, 5, 6])
+        for mem_format in [torch.channels_last, torch.channels_last_3d, torch.contiguous_format, torch.preserve_format]:
+            with self.assertRaisesRegex(RuntimeError, "memory format option is only supported by strided tensors"):
+                result = torch.zeros_like(sparse_tensor, memory_format=mem_format)
```

Comment:
I would add tests for zeros_like with a dense tensor but sparse layout request, and for a sparse tensor but a strided layout request. The first case should also error (although, as per a previous comment not for any memory-format-related reason) and the second case should work with memory_format. 

Ensuring zeros_like(dense -> sparse) errors will also ensure this test is updated properly if we ever change this code.